sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -720,6 +721,17 @@ class Analyzer(
     }
   }
 
+  // If we are resolving relations insides views (i.e. `AnalysisContext.catalogAndNamespace` is not
+  // empty), we need to expand single-part relation names with the current catalog and namespace of
+  // when the view was created.
+  private def expandRelationName(nameParts: Seq[String]): Seq[String] = {
+    if (nameParts.length == 1 && AnalysisContext.get.catalogAndNamespace.nonEmpty) {
+      AnalysisContext.get.catalogAndNamespace :+ nameParts.head
+    } else {
+      nameParts
```

Comment:
Is it possible nameParts.length is 2? Do we need to expand it with current catalog when the view was created?

Reply:
If nameParts contains namespace, but no catalog?

Reply:
This is a good question! In this case, we should extend the name parts with the catalog name.

But we should only do it if the first name part doesn't refer to a catalog.

Reply:
Yes, I think so. If the first part is not resolved to a catalog, we should expand it with current catalog when the view was created.